### PR TITLE
Delete preservation system agent from database

### PR DESCRIPTION
**Pull request by sevein**
This pull request introduces a new `Agent.objects.extend_queryset_with_preservation_system` method used to extend agent look-up iterables when we want to include the preservation system agent. This allows us to stop maintaining the preservation system agent as an actual database object. Most of the work was in updating our tests which had their own logic to inject the old database-based agent.

Connects to https://github.com/archivematica/Issues/issues/383.

### CODE REVIEW

**Review by replaceafill — APPROVED** [review]
:clap:

File: src/MCPClient/tests/test_sanitize.py
Comment (sevein) [inline src/MCPClient/tests/test_sanitize.py:345]: CC @replaceafill - hopefully not needed anymore. I'll be merging soon, as soon as I get the approval. We asked Evelyn to take a look at the METS.

**Pull request merged**
